### PR TITLE
Chunk Store implemented

### PR DESCRIPTION
**Pull request by ustulation**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,3 +9,20 @@ repository = "https://github.com/dirvine/maidsafe_vault"
...
+[dependencies]
+rustc-serialize = "*"
+cbor = "*"
+bchannel = "*"
+lru-cache = "*"
+time = "*"
+
+[dependencies.maidsafe_types]
+git = "https://github.com/dirvine/maidsafe_types"
+
```
Comment (maqi) [inline Cargo.toml]: need to include routing as well

File: src/vault.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
+    available at: http://www.maidsafe.net/licenses
+    Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
+    under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+    OF ANY KIND, either express or implied.
+    See the Licences for the specific language governing permissions and limitations relating to
+    use of the MaidSafe
+    Software.                                                                 */
+
+
```
Comment (maqi) [inline src/vault.rs]: this shall be the place implement facade as in C++ code base vault.cc

File: src/chunk_store.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    let mut return_val: Vec<u8> = Vec::new();
+
+    for it in self.entries.iter() {
+      if it.name == name {
+        let mut iv: Vec<u8> = Vec::with_capacity(AES256_IVSize);
+        let mut key: Vec<u8> = Vec::with_capacity(AES256_KeySize);
+
+        for it in name.iter().take(AES256_KeySize) {
+          key.push(*it);
+        }
```
Comment (maqi) [inline src/chunk_store.rs]: probably you can do key.push_all(name.iter().take(key_size)))

File: src/vault.rs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+use std::default::Default;
+
+
+use pmid_node::PmidNode;
+use self::Routing::Authority;
+use self::Routing::DhtIdentity;
+use self::Routing::Action;
+use self::Routing::RoutingError;
+
+impl Routing::Facade for PmidNode {
```
Comment (maqi) [inline src/vault.rs]: I think it shall be something like this : 
struct Vault {
  pmid_node : PmidNode;
  data_manager : DataManager;
  ...
  routing_node : RoutingNode
}

impl Vault {
  fn new() {
     routing_node : RoutingNode { traits : {|| self.handle_get(_)}}
  }
  fn handle_get(&self, our_authority: Authority, from_authority: Authority, from_address: DhtIdentity, data: Vec<u8>)->Result<Action, RoutingError> {
    let get_result = pmid_node.get(data);
    Ok(Action::Reply(get_result))
  }
}

File: src/vault.rs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+extern crate bchannel;
+
+use std::net::{TcpStream};
+use sodiumoxide::crypto;
+use std::sync::mpsc;
+use std::sync::mpsc::{Sender, Receiver};
+use std::default::Default;
+
+
+use pmid_node::PmidNode;
```
Comment (maqi) [inline src/vault.rs]: as this implementation will also depends on the implementation of routing_node in routing,
I will suggest leave this vault.rs as empty to make this PR only includes the work of chunk_store

File: src/chunk_store.rs (hunk 3 above)
Comment (ustulation) [inline src/chunk_store.rs]: didn't work (atleast not out of the box)

File: Cargo.toml (hunk 1 above)
Comment (ustulation) [inline Cargo.toml]: done

**Pull request merged**